### PR TITLE
filter instantiation at output

### PR DESCRIPTION
**Pull request by sugolov**
created filter input for the user, will work on output and importing schedules

### CODE REVIEW

**Comment by sugolov** [comment]
mistyped the commit description, meant to say at input

**Comment by kenneth-miura** [comment]
bro swap the 0/1 & N/Y to 1/0 & Y/N

File: src/main/java/controllers/CommandLineInterface.java [hunk 1]
```diff
@@ -34,7 +37,207 @@ public CommandLineInterface() {}
...
+
+    public static List<Filter> promptUserFilters(List<Course> userCourses) {
+        ArrayList<Filter> userFilters = new ArrayList<>();
+
+        Scanner scanner = new Scanner(System.in);
+        System.out.println("Would you like to configure any criteria for your schedules?" +
+                "\n" + "1 - Time conflicts" +
+                "\n" + "2 - Delivery method" +
+                "\n" + "3 - Enforce a time gap between courses" +
+                "\n" + "4 - Enforce times when you have no courses" +
```
Comment (kenneth-miura) [inline src/main/java/controllers/CommandLineInterface.java:51]: I'd say "Choose the time range in which you want to have courses"

File: src/main/java/controllers/CommandLineInterface.java [hunk 2]
```diff
@@ -34,7 +37,207 @@ public CommandLineInterface() {}
...
+        }
+
+        return userFilters;
+    }
+
+    private static List<Filter> promptTimeConflictFilter(){
+        Scanner scanner = new Scanner(System.in);
+        ArrayList<Filter> newFilters = new ArrayList<>();
+        System.out.println("Would you like to allow time conflicts between your courses? \n" +
+                "Enter 0/1 for N/Y. \n" +
```
Comment (kenneth-miura) [inline src/main/java/controllers/CommandLineInterface.java:94]: bro swap

**Review by kenneth-miura — CHANGES_REQUESTED** [review]
Looks good except some small user experience stuff

File: src/main/java/controllers/CommandLineInterface.java [hunk 3]
```diff
@@ -1,7 +1,10 @@
 package controllers;
```
Comment (EclipseIsDead) [inline src/main/java/controllers/CommandLineInterface.java:1]: double check spotless, some spacing is a lil sus

**Review by EclipseIsDead — APPROVED** [review]
LGTM, other than kenneth's ui changes

File: src/main/java/controllers/CommandLineInterface.java (hunk 3 above)
Comment (sugolov) [reply src/main/java/controllers/CommandLineInterface.java:1]: no!

File: src/main/java/controllers/CommandLineInterface.java (hunk 2 above)
Comment (sugolov) [reply src/main/java/controllers/CommandLineInterface.java:151]: yeah

**Review by roryhofland — APPROVED** [review]
kenneth just about covered it, looks good

**Review by kenneth-miura — APPROVED** [review]
looks good

**Comment by sugolov** [comment]
that's it for tongiht. i am happy with this. please review and merge ASAP.

**Review by kenneth-miura — APPROVED** [review]
yeah looks good

File: src/main/java/workers/Scheduler.java [hunk 4]
```diff
@@ -143,12 +108,14 @@ public Schedule createBasicSchedule(ArrayList<Course> courses) {
 
         if (courseTutorials.size() == 0) {
             for (Section lec : courseLectures) {
-                Schedule tempSchedule = new Schedule();
+                Schedule tempSchedule = s.clone();
```
Comment (EclipseIsDead) [inline src/main/java/workers/Scheduler.java:111]: yeah so later we need to make sure this isn't going to screw us later

**Review by EclipseIsDead — APPROVED** [review]
Love the UI, we will do something about clone later

**Pull request merged**